lib/TPP/Dialect/Tpp/TppUtils.cpp

```diff
@@ -334,24 +334,67 @@ static bool isBinaryOp(linalg::GenericOp linalgOp) {
   return false;
 }
 
-// Return true if the operands are all the same.
-static bool allOperandsHaveSameType(linalg::GenericOp linalgOp) {
-  Type operandType = linalgOp->getOperand(0).getType();
-  for (Value operand : linalgOp->getOperands()) {
-    Type currentType = operand.getType();
-    if (currentType != operandType)
+// Returns true if all the operands of the current linalgOp are shape type and
+// have the same statically-known shape. For example: A: memref<56x32xf32,
+// strided<[32, 1], offset: ?>> B: memref<56x32xf32> allOperandsHaveSameShape(A,
+// B) return true. C: memref<1x32xf32> allOperandsHaveSameShape(A, C) return
+// false.
+bool allOperandsHaveSameShapeAndStrides(TypeRange types) {
+  assert(types.size() > 1 && "expect one or more types");
+  if (!types[0].isa<MemRefType>())
+    return false;
+  auto firstOperandType = types[0].cast<MemRefType>();
+
+  // Step1. Validate rank.
+  int64_t rankOperand = firstOperandType.getRank();
+  for (Type currentType : types) {
+    if (!currentType.isa<MemRefType>())
+      return false;
+    if (currentType.cast<MemRefType>().getRank() != rankOperand)
+      return false;
+  }
+
+  // Step2. Validate shape, and strides.
+  // Get stride and offset for the first operand.
+  int64_t offsetFirstOperand = 0;
+  SmallVector<int64_t> stridesFirstOperand;
+  if (failed(getStridesAndOffset(firstOperandType, stridesFirstOperand,
+                                 offsetFirstOperand)))
+    return false;
+  ArrayRef<int64_t> shapeOperand = firstOperandType.getShape();
+  for (Type currentOperandType : types) {
+    // Compare the shape.
+    ArrayRef<int64_t> shapeCurrentOperand =
+        currentOperandType.cast<MemRefType>().getShape();
+    if (shapeCurrentOperand != shapeOperand)
+      return false;
+    int64_t offsetCurrentOperand = 0;
+    SmallVector<int64_t> stridesCurrentOperand;
+    // Compare the strides.
+    if (failed(getStridesAndOffset(currentOperandType.cast<MemRefType>(),
+                                   stridesCurrentOperand,
+                                   offsetCurrentOperand)))
+      return false;
+    if (stridesFirstOperand != stridesCurrentOperand)
```

Comment:
Shouldn't we also care about offsets?

Reply:
But shouldn't all the offsets start at the same place? Otherwise, the TPP call could segfault.

Reply:
Not necessary, we take into account the offset when computing the starting pointer for LIBXSMM. It is perfectly fine to iterate at different position in the tensor.

Reply:
This is guaranteeing that the shapes and strides are the same, if the offsets are different, there's something wrong.

Reply:
You suggest to compare also the offset?

Reply:
Yes, if possible. 

If they're dynamic, I'm not sure what to do because I don't know where this is coming from. If we are creating those, then it gets tricky, but if not, perhaps can we just fail on dynamic offsets?

Reply:
Discussed offline, the offset is introduced by the subviews we create, so should be "safe". Once we start adding TPP ops from non-linalg regions we can re-think the approach and start checking the offsets.